### ISSUE TITLE
OEP-32 has been superseded and should be updated 

### ISSUE THREAD

**Issue by e0d** [openedx/open-edx-proposals#407]
The guidance for external ids in OEP-32 is currently incorrect and that OEP has been superseded by this [ADR](https://github.com/openedx/edx-platform/blob/e1c8fb51bec2be3aa8da9750cb52b01728d1a740/openedx/core/djangoapps/external_user_ids/docs/decisions/0001-externalid.rst).

The OEP needs a significant rework.

**Comment by robrap** [comment]
Do you think the guidance is still correct for ids internal to the platform, and this is just about correcting guidance for external ids?

**Comment by e0d** [comment]
The ADR only covers the case of external IDs.  So, job one would be correcting that.

My preference would be that auto increment database IDs were not used as identifiers, but that's a much bigger, breaking, change I believe.

**Comment by robrap** [comment]
I agree with your preference, but the point of the OEP was to codify what to use, given the reality of our situation. Better than using user id in some cases, and username in other cases, as was happening.

**Comment by sarina** [comment]
@robrap @e0d - is [OEP-32](https://docs.openedx.org/projects/openedx-proposals/en/latest/architectural-decisions/oep-0032-arch-unique-identifier-for-users.html) completely incorrect and should be marked as Obsolete, or is it simply that parts need updating?

**Comment by robrap** [comment]
It is partially correct. It is valid for internal ids, and invalid for external ids. I'm not sure if there are legacy exceptions to the external ids case that do use this same internal id.

**Comment by sarina** [comment]
Thanks. Assuming https://github.com/openedx/open-edx-proposals/pull/586 passes, I will mark this OEP as Needs Revision and point to this thread.

**Comment by sarina** [comment]
Put OEP-32 in Needs Revision status, please take a look at #603 - this ticket will be kept open for discussion on how to actually update the OEP for real

**Comment by sarina** [comment]
Updating Feb 2025 as part of my OEP cleanup. This issue still needs to be addressed. I don't feel I have the context or ability to do so. Is there anyone I could tag who might have the context & capacity to make this update?

**Comment by robrap** [comment]
@sarina: I added some comments to https://github.com/openedx/open-edx-proposals/pull/603 before realizing it was approved/merged. We could consider updating again.

Other than that, I'm not sure how urgent/important it is to turn the note you added into the OEP into a more complete update. I'll leave that to @e0d to decide, and determine whether to leave open this issue. I imagine the note is good enough for quite a while, so I don't feel any burning desire to do the update myself.
